Rancher Desktop 1.8.1 darwin(x64) fails to startup on Apple Silicon M2 Max

### Actual Behavior

The following error in lima.log presents itself when Rancher Desktop is getting initialized.

```
2023-06-03T10:43:18.577Z: Lima: executing: sudo rm -f /tmp/rd-nerdctl-EfiRzz.nerdctl: Error: spawn /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl.ventura EACCES
2023-06-03T10:43:18.577Z: Error trying to start/update containerd: Error: spawn /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl.ventura EACCES:  Error: spawn /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl.ventura EACCES
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:478:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -13,
  code: 'EACCES',
  syscall: 'spawn /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl.ventura',
  path: '/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl.ventura',
  spawnargs: [
    '--debug',
    'shell',
    '--workdir=.',
    '0',
    'sudo',
    'rm',
    '-f',
    '/tmp/rd-nerdctl-EfiRzz.nerdctl'
  ]
}
```

<img width="758" alt="image" src="https://github.com/rancher-sandbox/rancher-desktop/assets/19311617/d737971c-430f-483c-8a8b-2b83a7de3093">



### Steps to Reproduce

Install Rancher Desktop on M2 machine first time

### Result

<img width="758" alt="image" src="https://github.com/rancher-sandbox/rancher-desktop/assets/19311617/d737971c-430f-483c-8a8b-2b83a7de3093">

### Expected Behavior

Rancher Desktop should be able to setup correctly

### Additional Information

_No response_

### Rancher Desktop Version

1.8.1

### Rancher Desktop K8s Version

1.27.1

### Which container engine are you using?

moby (docker cli)

### What operating system are you using?

macOS

### Operating System / Build Version

macOS Ventura 13.4

### What CPU architecture are you using?

arm64 (Apple Silicon)

### Linux only: what package format did you use to install Rancher Desktop?

None

### Windows User Only

_No response_

error	00:29:27.983222+0530	kernel	mcxalr{310} [ERROR] msleep waiting for client reply (uid:502 seqid:19223 path:/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl.ventura) failed (4).  Result: -1
error	00:29:27.983233+0530	kernel	mcxalr{311} [ERROR] Unable to check with client for uid=502  path=/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl.ventura
default	00:29:27.983236+0530	kernel	mcxalr{312} ** Denying execute for uid=502  path=/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl.ventura

--- a/pkg/rancher-desktop/assets/lima-config.yaml
+++ b/pkg/rancher-desktop/assets/lima-config.yaml
@@ -168,6 +168,24 @@ provision:
     mkcert -install
     mkcert localhost
     chown -R nobody:nobody $CAROOT
+
+    # Get the hash of the newly installed certificate
+    CA_SERIAL=$(openssl x509 -in "${CAROOT}"/rootCA.pem -noout -serial | sed 's/serial=//' | xargs -I{} echo "ibase=16; {}" | bc)
+    CA_SUBJECT_HASH=$(openssl x509 -noout -subject_hash -in /usr/local/share/ca-certificates/mkcert_development_CA_"${CA_SERIAL}".crt)
+    # Remove old mkcert certificates
+    for cert in /usr/local/share/ca-certificates/mkcert_development_CA_*.crt; do
+      cert_name=$(basename "$cert")
+      if [ -f "$cert" ] && [ "$cert_name" != "mkcert_development_CA_${CA_SERIAL}.crt" ]; then
+        rm -f "$cert"
+      fi
+    done
+    rm -f /etc/ssl/certs/"${CA_SUBJECT_HASH}".*
+    # Remove any orphaned symlinks with old naming pattern
+    for symlink in /etc/ssl/certs/ca-cert-mkcert_development_CA_*.pem; do
+      if [ -L "$symlink" ] && [ ! -e "$symlink" ]; then
+        rm -f "$symlink"
+      fi
+    done
 - # Configure HTTPS_PROXY to OpenResty
   mode: system
   script: |

--- a/pkg/rancher-desktop/assets/lima-config.yaml
+++ b/pkg/rancher-desktop/assets/lima-config.yaml
@@ -165,27 +165,11 @@ provision:
     export CAROOT=/run/mkcert
     mkdir -p $CAROOT
     cd $CAROOT
+    # Remove old mkcert certificates
+    rm -f /usr/local/share/ca-certificates/mkcert_development_CA_*.crt
     mkcert -install
     mkcert localhost
     chown -R nobody:nobody $CAROOT
-
-    # Get the hash of the newly installed certificate
-    CA_SERIAL=$(openssl x509 -in "${CAROOT}"/rootCA.pem -noout -serial | sed 's/serial=//' | xargs -I{} echo "ibase=16; {}" | bc)
-    CA_SUBJECT_HASH=$(openssl x509 -noout -subject_hash -in /usr/local/share/ca-certificates/mkcert_development_CA_"${CA_SERIAL}".crt)
-    # Remove old mkcert certificates
-    for cert in /usr/local/share/ca-certificates/mkcert_development_CA_*.crt; do
-      cert_name=$(basename "$cert")
-      if [ -f "$cert" ] && [ "$cert_name" != "mkcert_development_CA_${CA_SERIAL}.crt" ]; then
-        rm -f "$cert"
-      fi
-    done
-    rm -f /etc/ssl/certs/"${CA_SUBJECT_HASH}".*
-    # Remove any orphaned symlinks with old naming pattern
-    for symlink in /etc/ssl/certs/ca-cert-mkcert_development_CA_*.pem; do
-      if [ -L "$symlink" ] && [ ! -e "$symlink" ]; then
-        rm -f "$symlink"
-      fi
-    done
 - # Configure HTTPS_PROXY to OpenResty
   mode: system
   script: |

--- a/pkg/rancher-desktop/assets/scripts/configure-allowed-images
+++ b/pkg/rancher-desktop/assets/scripts/configure-allowed-images
@@ -14,6 +14,8 @@ adduser -S -D -H -h /dev/null -s /sbin/nologin -u 65534 -G nobody -g nobody nobo
 export CAROOT=/run/mkcert
 mkdir -p $CAROOT
 cd $CAROOT
+# Remove old mkcert certificates
+rm -f /usr/local/share/ca-certificates/mkcert_development_CA_*.crt
 mkcert -install
 mkcert localhost
 chown -R nobody:nobody $CAROOT

